### ISSUE TITLE
Pyright 1.1.340 doesn't include the scheme in the URI for the rename LSP action

### ISSUE THREAD

**Issue by tmke8** [microsoft/pyright#6725]
It works fine with 1.1.339.

When using the rename functionality in neovim ([vim.lsp.buf.rename](https://neovim.io/doc/user/lsp.html#vim.lsp.buf.rename()), so, `textDocument/rename`) with the pyright LS, I get this error:

```
URI must contain a scheme: /home/tmke8/to/my/project/main.py
```

I'm guessing it's missing `file://`? If it helps I can try to investigate the problem further.

**VS Code extension or command-line**
As language server with neovim.

**Comment by rchiodo** [comment]
Thanks for the issue. Can you outline the repro steps more? I'm not familiar with neovim. I'm really not sure how to get pyright into neovim as it doesn't seem to be listed in the neovim plugins:
https://dotfyle.com/neovim/plugins/trending?page=1&q=pyright

**Comment by erictraut** [comment]
@tmke8, are you using the [coc-pyright](https://github.com/fannheyward/coc-pyright) extension for vim? Could you provide the full JSON for the LSP command that is being sent to the pyright language server?

My hypothesis is that the command includes a malformed URI, and older versions of pyright were permissively handling it. If my hypothesis is correct, then you'll need to work with the maintainers of coc-pyright to get this fixed.

**Comment by tmke8** [comment]
I'm not using coc-pyright, I'm using the standard configuration from https://github.com/neovim/nvim-lspconfig :

```lua
require('lspconfig').pyright.setup {}
```
and then
```lua
vim.keymap.set('n', '<space>rn', vim.lsp.buf.rename, opts)
```
somewhere else in the neovim config file.

I'll see whether I can log the JSON somehow.

**Comment by tmke8** [comment]
<details><summary>These should be the logs:</summary>

```
[DEBUG][2023-12-13 18:53:54] .../vim/lsp/rpc.lua:284	"rpc.send"	{
  id = 8,
  jsonrpc = "2.0",
  method = "textDocument/rename",
  params = {
    newName = "ACSSurveyYear",
    position = {
      character = 6,
      line = 22
    },
    textDocument = {
      uri = "file:///home/tmk/dev/python/conduit/conduit/fair/data/datasets/acs.py"
    }
  }
}
[DEBUG][2023-12-13 18:53:54] .../vim/lsp/rpc.lua:387	"rpc.receive"	{
  id = 8,
  jsonrpc = "2.0",
  result = {
    documentChanges = { {
        edits = { {
            annotationId = "default",
            newText = "ACSSurveyYear",
            range = {
              ["end"] = {
                character = 89,
                line = 19
              },
              start = {
                character = 79,
                line = 19
              }
            }
          }, {
            annotationId = "default",
            newText = "ACSSurveyYear",
            range = {
              ["end"] = {
                character = 16,
                line = 22
              },
              start = {
                character = 6,
                line = 22
              }
            }
          }, {
            annotationId = "default",
            newText = "ACSSurveyYear",
            range = {
              ["end"] = {
                character = 38,
                line = 106
              },
              start = {
                character = 28,
                line = 106
              }
            }
          } },
        textDocument = {
          uri = "/home/tmk/dev/python/conduit/conduit/fair/data/datasets/acs.py"
        }
      } }
  }
}
```

</details>

**Comment by erictraut** [comment]
@tmke8, thanks, those logs are really helpful.

@rchiodo, yeah, it looks like this is a regression related to your URI change. I'm guessing the problem is somewhere in `_convertToWorkspaceEditWithDocumentChanges`.

**Comment by erictraut** [comment]
@rchiodo, is it OK to tag this as "addressed in next version"?

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.341, which I just published.

**Comment by tmke8** [comment]
I can confirm that it works again in 1.1.341.
